curvlinops/kfac.py

```diff
@@ -346,37 +434,22 @@ def _matmat(self, M: List[Tensor]) -> List[Tensor]:
             ):
                 w_pos, b_pos = param_pos["weight"], param_pos["bias"]
                 # v denotes the free dimension for treating multiple vectors in parallel
-                M_w = rearrange(M[w_pos], "c_out ... v -> c_out (...) v")
+                M_w = rearrange(M[w_pos], "m ... v -> m (...) v")
```

Comment:
I think this line allows us to remove the same `if` condition in `_left_and_right_multiply` [above](https://github.com/f-dangel/curvlinops/pull/173/files#diff-81987851ce797c96eec5b13c693f28ffd852d7a3af0ab1e712d5da58a02a9ed9R380-R381).

Reply:
Note: We can also replace some `...` in `_left_and_right_multiply` with a single letter, which should be safer.

Reply:
Unless I'm missing something the line appears twice because they are in separate code paths: the line above is called before `_left_and_right_multiply`. However, the line you [reference](https://github.com/f-dangel/curvlinops/pull/173/files#diff-81987851ce797c96eec5b13c693f28ffd852d7a3af0ab1e712d5da58a02a9ed9R380-R381) is in `_separate_left_and_right_multiply`.

> Note: We can also replace some ... in _left_and_right_multiply with a single letter, which should be safer.

`_left_and_right_multiply` doesn't have any `...` and I think the `...` in `_separate_left_and_right_multiply` are necessary because of what I said in my comment [here](https://github.com/f-dangel/curvlinops/pull/173#discussion_r1965325561).